modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleList/RuleList.es.js

```diff
@@ -372,98 +364,265 @@ class RuleList extends Component {
-		contains: Liferay.Language.get('contains'),
-		'due-to-missing-fields': Liferay.Language.get('due-to-missing-fields'),
-		'equals-to': Liferay.Language.get('is-equal-to'),
-		'greater-than': Liferay.Language.get('is-greater-than'),
-		'greater-than-equals': Liferay.Language.get(
-			'is-greater-than-or-equal-to'
-		),
-		'is-empty': Liferay.Language.get('is-empty'),
-		'less-than': Liferay.Language.get('is-less-than'),
-		'less-than-equals': Liferay.Language.get('is-less-than-or-equal-to'),
-		'not-contains': Liferay.Language.get('does-not-contain'),
-		'not-equals-to': Liferay.Language.get('is-not-equal-to'),
-		'not-is-empty': Liferay.Language.get('is-not-empty'),
-	}),
-};
-
-Soy.register(RuleList, templates);
-
-export default RuleList;
+/**
+ * Prints the DDM form card rule.
+ */
+const EmptyList = ({message}) => {
+    return (
+        <div className="main-content-body">
+            <div className="sheet taglib-empty-result-message">
+                <div className="taglib-empty-result-message-header"></div>
+
+                {message && (
+                    <div className="sheet-text text-center text-muted">
+                        <p className="text-default">{message}</p>
+                    </div>
+                )}
+            </div>
+        </div>
+    )
+}
+
+/**
+ * Prints the DDM form card rule.
+ */
+const Label = ({content}) => {
+    return (
+        <span className="label label-lg label-secondary" data-original-title={content} title={content}>
+            <span className="text-truncate-inline">
+                <span className="text-truncate">{content}</span>
+            </span>
+	    </span>
+    )
+}
+
+
+/**
+ * Prints Rules Conditions.
+ */
+const Condition = ({
+    operandType,
+    operandLabel,
+    operandValue
+}) => {
+    const SubCondition = ({operandType}) => {
+        if(['double', 'integer', 'json', 'option', 'string', 'text', 'numeric'].includes(operandType)){
+            return  (<span>{Liferay.Language.get('value')} </span>);
+        }
+    
+        if(operandType === 'user'){
+            return (<span>{Liferay.Language.get('user')} </span>);
+        }
+
+        if(operandType === 'field'){
+            return (<span>{Liferay.Language.get('field')} </span>);
+        }
+
+        if(operandType === 'list'){
+            return (<span>{Liferay.Language.get('list')} </span>);
+        }
+
+        return null;
+    }
+
+    return (
+        <div className="ddm-condition-container">
+            <SubCondition operandType={operandType}/>
+
+            <Label content={operandLabel || operandValue} />
+	    </div>
+    );
+}
+
+/**
+ * Prints the show action.
+ */
+const Action = ({
+    action,
+    dataProviderName,
+    outputLabel,
+}) => {
+    if(action === 'auto-fill'){
+        return (
+            <>
+                {outputLabel.map(output => {
+                    <>
+                        <Label content={output} />
+                        {!isLast(output) ? ',' : ''}
```

Comment:
isLast is defined everywhere on the code, just leaving a comment for reminding to get rid this

Reply:
you are right, I need to replace it for a real function.